File: 251001/Stalbou/publisher/src/main/java/org/ex/distributed_computing/mapper/NoticeMapper.java
package org.ex.distributed_computing.mapper;

import java.util.List;
import org.ex.distributed_computing.dto.request.NoticeRequestDTO;
import org.ex.distributed_computing.dto.response.NoticeResponseDTO;
import org.ex.distributed_computing.model.Notice;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface NoticeMapper {

  @Mapping(source = "newsId", target = "news.id")
  Notice toEntity(NoticeRequestDTO dto);

  @Mapping(source = "news.id", target = "newsId")
  NoticeResponseDTO toDto(Notice notice);

  NoticeResponseDTO convert(NoticeRequestDTO dto);

  List<NoticeResponseDTO> toDtoList(List<Notice> notices);
}

File: 251001/Voytov/discussion/src/main/kotlin/org/ex/discussion/mapper/NoticeMapper.kt
package org.ex.discussion.mapper

import org.ex.discussion.dto.request.NoticeRequestDTO
import org.ex.discussion.dto.response.NoticeResponseDTO
import org.ex.discussion.model.Notice
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants.ComponentModel

@Mapper(
    componentModel = ComponentModel.SPRING,
    injectionStrategy = InjectionStrategy.CONSTRUCTOR
)
interface NoticeMapper {

    fun toDto(entity: Notice): NoticeResponseDTO

    fun toEntity(dto: NoticeRequestDTO): Notice
}